src/ui/DynamicFacet/DynamicFacetValues/DynamicFacetValueDuplicateResolver.ts

```diff
@@ -0,0 +1,37 @@
+import { IFacetResponseValue } from '../../../rest/Facet/FacetResponse';
+import { FacetValueState } from '../../../rest/Facet/FacetValueState';
+
+export function filterDuplicateFacetValues(values: IFacetResponseValue[]) {
+  let resolvedValues: IFacetResponseValue[] = [];
+  const map: Record<string, IFacetResponseValue[]> = {};
+
+  values.forEach(facetValue => {
+    const value = facetValue.value;
+    const lowercase = value.toLowerCase();
+
+    const entry = map[lowercase];
+
+    map[lowercase] = entry ? entry.concat(facetValue) : [facetValue];
+  });
+
+  Object.keys(map).forEach(key => {
+    const collection = map[key];
+
+    if (collection.length < 2) {
+      resolvedValues = resolvedValues.concat(collection);
+      return;
+    }
+
+    const sorted = collection.sort((a, b) => b.numberOfResults - a.numberOfResults);
```

Comment:
Aren't we completely overriding the sorting done by the API with this line here, or am I misreading ?

Reply:
The descending sort here is just on the duplicates to figure out the source of truth. The value with incorrect casing has a count of 0, so it will end up at the end of the array.